test/iaas.GcpClient.spec.js

```diff
@@ -0,0 +1,538 @@
+  }],
+  response: undefined,
+  message: 'my-test-user@my-gcp-dev.iam.myaccounts.com does not have storage.objects.delete access'
+};
+let deleteSnapshotEventHandlers = {};
+const deleteSnapshotResponse = [{
+    on: (event, callback) => {
+      _.set(deleteSnapshotEventHandlers, event, callback);
+      return true;
+    },
+    removeAllListeners: () => {
+      deleteSnapshotEventHandlers = {};
+      return true;
+    }
+  },
+  {
+    kind: 'compute#operation',
+    id: '333333',
+    name: 'operation-xxxxxx',
+    operationType: 'delete',
+    targetLink: 'https://myaccounts.com/compute/v1/projects/my-gcp-dev/global/snapshots/snapshot-1',
+    targetId: '11111',
+    status: 'PENDING',
+    user: 'my-test-user@my-gcp-dev.iam.myaccounts.com',
+    progress: 0,
+    insertTime: '2018-03-15T08:31:05.592-07:00',
+    selfLink: 'https://myaccounts.com/compute/v1/projects/my-gcp-dev/global/operations/operation-xxxxxx'
+  }
+];
+let streamEventHandlers = {};
+const jsonContent = '{"content": "This is a sample content"}';
+const streamResponse = {
+  on: (event, callback) => {
+    _.set(streamEventHandlers, event, callback);
+    return true;
+  },
+  removeListener: (event, callback) => {
+    _.unset(streamEventHandlers, event, callback);
+    return true;
+  },
+  end: () => {
+    return true;
+  }
+};
+const utilsStreamToPromiseStub = function streamToPromise(stream) {
+  return new Promise((resolve, reject) => {
+    stream.on('end', () => {
+      resolve(jsonContent);
+    });
+    stream.on('error', (err) => {
+      reject(err);
+    });
+  });
+};
+const fileStub = {
+  delete: () => {
+    return Promise.resolve(deleteFileSuccessResponse);
+  },
+  createWriteStream: () => {
+    return streamResponse;
+  },
+  createReadStream: () => {
+    return streamResponse;
+  }
+};
+const fileNotFoundStub = {
+  delete: () => {
+    return Promise.reject(fileNotFoundResponse);
+  }
+};
+const fileForbiddenStub = {
+  delete: () => {
+    return Promise.reject(fileForbiddenResponse);
+  }
+};
+const validBlobName = 'blob1.txt';
+const notFoundBlobName = 'blob2.txt';
+const bucketStub = {
+  get: () => {
+    return Promise.resolve(bucketMetadataResponse);
+  },
+  getFiles: () => {
+    return Promise.resolve(getFilesResponse);
+  },
+  file: (file) => {
+    if (file === validBlobName) {
+      return fileStub;
+    } else if (file === notFoundBlobName) {
+      return fileNotFoundStub;
+    } else {
+      return fileForbiddenStub;
+    }
+  }
+};
+const snapshotStub = {
+  delete: () => {
+    return Promise.resolve(deleteSnapshotResponse);
+  }
+};
+const deleteSnapshotErrMsg = 'Delete snapshot failed';
+const snapshotFailedStub = {
+  delete: () => {
+    return Promise.reject(deleteSnapshotErrMsg);
+  }
+};
+
+describe('iaas', function () {
+  describe('GcpClient', function () {
+    let sandbox, client;
+    before(function () {
+      sandbox = sinon.sandbox.create();
```

Comment:
Better to create sandbox and restore sandbox where its used. Here at first look it seemed like this was not used.

Reply:
Done.